openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2897,132 +2904,233 @@ class GBSAParameterHandler(ParameterHandler):
+                                                 f"A (data source specified value of "
+                                                 f"{self.solvent_radius})")
+
+
+        # If we're using OBC2 via GBSAOBCForce, then we need to ensure that
+        #   solvent_radius is 1.4 A
+        # Justification at https://github.com/openforcefield/openforcefield/pull/363)
+        if self.gb_model == 'OBC2':
+
+            if self.solvent_radius != 1.4 * unit.angstrom:
+                raise IncompatibleParameterError(f"The current implementation of OBC1 GBSA does not "
+                                                 f"support solvent_radius values other than 1.4 "
+                                                 f"A (data source specified value of "
+                                                 f"{self.solvent_radius})")
+
+
+
+    # Tolerance when comparing float attributes for handler compatibility.
+    _SCALETOL = 1e-5
+
+    def check_handler_compatibility(self,
+                                    other_handler):
+        """
+        Checks whether this ParameterHandler encodes compatible physics as another ParameterHandler. This is
+        called if a second handler is attempted to be initialized for the same tag.
+
+        Parameters
+        ----------
+        other_handler : a ParameterHandler object
+            The handler to compare to.
+
+        Raises
+        ------
+        IncompatibleParameterError if handler_kwargs are incompatible with existing parameters.
+        """
+        float_attrs_to_compare = ['solvent_dielectric', 'solute_dielectric']
+        string_attrs_to_compare = ['gb_model', 'sa_model']
+        unit_attrs_to_compare = ['surface_area_penalty', 'solvent_radius']
+
+        self._check_attributes_are_equal(other_handler, identical_attrs=string_attrs_to_compare,
+                                         tolerance_attrs=float_attrs_to_compare+unit_attrs_to_compare,
+                                         tolerance=self._SCALETOL)
+
+
+
+    def create_force(self, system, topology, **kwargs):
+        import simtk
+
+        self._validate_parameters()
+
+        # Grab the existing nonbonded force (which will have particle charges)
+        existing = [system.getForce(i) for i in range(system.getNumForces())]
+        existing = [
+            f for f in existing if type(f) == openmm.NonbondedForce
+        ]
+
+        nonbonded_force = existing[0]
+
+        # No previous GBSAForce should exist, so we're safe just making one here.
+        force_map = {
+            'HCT': simtk.openmm.app.internal.customgbforces.GBSAHCTForce,
+            'OBC1': simtk.openmm.app.internal.customgbforces.GBSAOBC1Force,
+            'OBC2': simtk.openmm.GBSAOBCForce,
+            # It's tempting to do use the class below, but the customgbforce
+            # version of OBC2 doesn't provide setSolventRadius()
+            #'OBC2': simtk.openmm.app.internal.customgbforces.GBSAOBC2Force,
```

Comment:
I'm not sure how I feel about splitting this into two separate class types. Does GBSAOBCForce not support the features needed for HCT/OBC1?

Reply:
GBSAOBCForce doesn't let us modify alpha/beta/gamma parameters (as far as I can tell). In some parts of the code, requesting an OBC2 model where kappa=0 yields a GBSAOBCForce [1], so its alpha/beta/gamma must be configured for OBC2. Thus I think  it's unsuitable to represent OBC1.
[1] https://github.com/pandegroup/openmm/blob/f9106ddb6a0d72c960c73b23ed0e07cd7190a80f/wrappers/python/simtk/openmm/app/internal/amber_file_parser.py#L985-L991